rspace-bench/src/main/scala/coop/rchain/rspace/bench/lmdb/LMDBOpsBench.scala

```diff
@@ -0,0 +1,369 @@
+      dbDir.recursivelyDelete()
+      ()
+    }
+  }
+
+  class BaseLMDBOpsBenchState {
+    implicit val codecC: Codec[Channel]       = serializeChannel.toCodec
+    implicit val codecP: Codec[Pattern]       = serializePattern.toCodec
+    implicit val codecA: Codec[Entry]         = serializeInfo.toCodec
+    implicit val codecK: Codec[EntriesCaptor] = serializeEntriesCaptor.toCodec
+
+    val mapSize: Long  = 1024L * 1024L * 1024L
+    val noTls: Boolean = false
+
+    var dbDir: Path              = _
+    var dbGnats: Dbi[ByteBuffer] = _
+    var dbJoins: Dbi[ByteBuffer] = _
+    var env: Env[ByteBuffer]     = _
+
+    val randomData = generate(100, 100)
+
+    def indexedKey(i: Int): ByteBuffer = {
+      val bytes = Array.fill[Byte](2)(Random.nextInt().toByte)
+      val mbb   = ByteBuffer.allocateDirect(bytes.length + 4)
+      mbb.put(bytes)
+      mbb.putInt(i)
+      println(mbb.getInt(2))
+      println(s"!!!(${mbb.get(0)}${mbb.get(1)})==${mbb.getInt(2)} ---- $i")
+      mbb.flip()
+      mbb
+    }
+
+    def populate(): Unit =
+      randomData.foreach(x => {
+        val txn = env.txnWrite()
+        dbGnats.put(txn, x._1, x._2)
+        txn.commit()
+      })
+
+    def setup(): Unit = {
+      dbDir = Files.createTempDirectory("rchain-rspace-mixed-bench-")
+      val branch = Branch.MASTER
+      val flags  = List(EnvFlags.MDB_NOTLS)
+      env = Context.env(dbDir, mapSize, flags)
+      dbGnats = env.openDbi(s"${branch.name}-gnats", MDB_CREATE)
+      dbJoins = env.openDbi(s"${branch.name}-joins", MDB_CREATE)
+      val trieStore = LMDBTrieStore
+        .create[Blake2b256Hash, GNAT[Channel, Pattern, Entry, EntriesCaptor]](env, dbDir)
+      store = new LMDBStore[Channel, Pattern, Entry, EntriesCaptor](
+        env,
+        dbDir,
+        dbGnats,
+        dbJoins,
+        trieStore,
+        branch
+      )
+      populate()
+    }
+
+    def generate(elems: Int, hashSize: Int): Seq[(ByteBuffer, ByteBuffer)] =
+      (0 to elems).map(_ => (key, hash(hashSize)))
+
+    def roundtrip(
+        data: Seq[(ByteBuffer, ByteBuffer)],
+        db: Dbi[ByteBuffer],
+        env: Env[ByteBuffer]
+    ): Seq[(ByteBuffer, ByteBuffer)] = {
+      data.foreach(x => {
+        val txn = env.txnWrite()
+        db.put(txn, x._1, x._2)
+        txn.commit()
+      })
+      data.map(x => {
+        val txn = env.txnRead()
+        val r   = (x._1, db.get(txn, x._1))
+        txn.commit()
+        r
+      })
+    }
+
+    def key: ByteBuffer = {
+      val bytes = Array.fill[Byte](32)(Random.nextInt().toByte)
+      val bb    = ByteBuffer.allocateDirect(bytes.length)
+      bb.put(bytes)
+      bb.flip()
+      bb
+    }
+
+    def hash(s: Int): ByteBuffer = {
+      val size: Int = 1024 * s * 10
+      val bytes     = Array.fill[Byte](size)(Random.nextInt().toByte)
+      val bb        = ByteBuffer.allocateDirect(bytes.length)
+      bb.put(bytes)
+    }
+
+    var store: LMDBStore[Channel, Pattern, Entry, EntriesCaptor] = _
```

Comment:
You're not using the store anywhere in benchmarks. Why create it in the first place? Just use pure `Dbi[ByteBuffer]`, no need to wrap it in the store. This should remove a lot of boilerplate here